src/cart/crt_iv.c

```diff
@@ -1141,6 +1141,8 @@ crt_ivf_rpc_issue(d_rank_t dest_node, crt_iv_key_t *iv_key,
 			IV_DBG(iv_key, "added to kip_entry=%p\n", entry);
 			D_MUTEX_UNLOCK(&entry->kip_lock);
 			D_MUTEX_UNLOCK(&ivns_internal->cii_lock);
+			if (rc == 0)
+				IVNS_DECREF(ivns_internal);
```

Comment:
that does not look correct. crt_ivf_pending_request_add() increments ivns passed to it, as it keeps a reference to it in the list.

the corresponding decref is done in crt_ivf_pending_reqs_process(). If there is some scenario that causes ref counts to be missed there, that would be a proper place to add it

Reply:
but the ivns to crt_ivf_rpc_issue() needs to be decref, since crt_ivf_pending_request_add() already increase the refcount, so it needs to decref here for rc == 0 case.

Reply:
yep as discussed in the IM offline, you are correct. good catch. Please add a comment here though indicating that this is decref for corresponding addref done prior to crt_ivf_rpc_issue()